Optimiser.fs

```diff
@@ -481,6 +493,15 @@ module Graph =
                                               | _ -> false))
                      ps)
 
+    /// Decides whether a given Command contains any `assume` command
+    /// in any of the sequentially composed primitives inside it
+    let hasAssume : Command -> bool =
+        fun c ->
```

Comment:
Could do `hasAssume (c : Command) : bool =` here, or would this be stylistically bad?